build/elasticsearch/Dockerfile

```diff
@@ -12,6 +12,7 @@ RUN \
     org.wikimedia.search:extra:${ELASTICSEARCH_PLUGIN_WIKIMEDIA_EXTRA} && \
   ./bin/elasticsearch-plugin install \
     org.wikimedia.search.highlighter:experimental-highlighter-elasticsearch-plugin:${ELASTICSEARCH_PLUGIN_WIKIMEDIA_HIGHLIGHTER}
+COPY healthcheck.sh /healthcheck.sh
```

Comment:
Could you tell me about your thinking why not to use the `HEALTHCHECK` dockerfile instruction? What are the downsides?

Reply:
It is taking the lead from nginx, mariadb, etc images. They don't build in a HEALTHCHECK instruction in the their Dockerfiles but expect it to be manually applied via the CLI or a Compose file. One of the reasons for that in Kubernetes use of the image the HEALTHCHECK will be ignored, which can make it misleading to have it in the file. The other reason, as I understand it, is that using a HEALTHCHECK instructions results in a lot of extra logging that otherwise doesn't happen in the Compose scenario? Basically I think there is a case against baking in the HEALTHCHECK in the Dockerfile itself, but it's not a terribly strong case--just a slightly stronger case than the reasons to choose to have it there. https://www.reddit.com/r/kubernetes/comments/15rnhu8/comment/jwkerd7/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button